src/backend/distributed/executor/executor.c

```diff
@@ -57,10 +57,16 @@ typedef struct DistributedExecution
 	List *sessionList;
 
 	/*
-	 * Flag to indiciate that the number of connections has changed
-	 * and the WaitEventSet needs to be rebuilt.
+	 * Flag to indiciate that the set of connections we are interested
+	 * in has changed and waitEventSet needs to be rebuilt.
 	 */
-	bool rebuildWaitEventSet;
+	bool connectionSetChanged;
+
+	/*
+	 * Flag to indiciate that the set of wait events we are interested
+	 * in has changed and waitEventSet needs to be updated.
+	 */
+	bool waitFlagsChanged;
```

Comment:
I think we set this flag somewhat liberally, since we don't check whether the flags actually changed, but that seems mostly fine because `ModifyWaitEvent` is a noop when the flags don't change. Maybe we can add a comment to that effect somewhere?

Reply:
done. update the comments above `waitFlagsChanged`.